File: examples/Lua/psenvpub.lua
--
--  Pubsub envelope publisher
--  Note that the zhelpers.h file also provides s_sendmore
--
--  Author: Robert G. Jakabosky <bobby@sharedrealm.com>
--
require"zmq"
require"zhelpers"

--  Prepare our context and publisher
local context = zmq.init(1)
local publisher = context:socket(zmq.PUB)
publisher:bind("tcp://*:5563")

while true do
    --  Write two messages, each with an envelope and content
    publisher:send("A", zmq.SNDMORE)
    publisher:send("We don't want to see this")
    publisher:send("B", zmq.SNDMORE)
    publisher:send("We would like to see this")
    s_sleep (1000)
end
--  We never get here but clean up anyhow
publisher:close()
context:term()

File: examples/Erlang/psenvpub.erl
#! /usr/bin/env escript
%%
%% Pubsub envelope publisher
%%

main(_) ->
    %% Prepare our context and publisher
    {ok, Context} = erlzmq:context(),
    {ok, Publisher} = erlzmq:socket(Context, pub),
    ok = erlzmq:bind(Publisher, "tcp://*:5563"),

    loop(Publisher),

    %% We never get here but clean up anyhow
    ok = erlzmq:close(Publisher),
    ok = erlzmq:term(Context).

loop(Publisher) ->
    %% Write two messages, each with an envelope and content
    ok = erlzmq:send(Publisher, <<"A">>, [sndmore]),
    ok = erlzmq:send(Publisher, <<"We don't want to see this">>),
    ok = erlzmq:send(Publisher, <<"B">>, [sndmore]),
    ok = erlzmq:send(Publisher, <<"We would like to see this">>),
    timer:sleep(1000),
    loop(Publisher).
